packages/wrangler/src/proxy.ts

```diff
@@ -313,10 +327,12 @@ export function usePreviewServer({
       });
 
     return () => {
-      proxyServer.close();
       abortController.abort();
+      // Running `proxy.server.close()` does not close open connections, preventing the process from exiting.
+      // So we use this `terminator` to close all the connections and force the server to shutdown.
+      proxy.terminator.terminate();
```

Comment:
The terminate w/ the previous `.abort()` addition is looking very promising to improve Wrangler graceful shutdown.